pull.go

```diff
@@ -173,6 +173,9 @@ func pullImage(ctx context.Context, store storage.Store, imageName string, optio
 	if err == nil {
 		return destRef, nil
 	}
+	if !useRegistries {
```

Comment:
So, the parameter does not say “registries should be used” at all, as the name suggests, but it affects how errors are reported. That’s confusing.

I don’t think the code below, which talks about short names and search, should be down here in `pullImage` in the first place: it belongs where the search behavior is implemented, i.e. in `resolveImage`.  At that point we can have a design discussion about what `resolveImage` should do on failure:
- maybe collect a list of all (registry, error) pairs, and report them all
- or maybe report the underlying error when exactly one registry is involved
- or maybe continue to do what it does now and just vaguely mention the search list